### ISSUE TITLE
Locale issues

### ISSUE THREAD

**Issue by WildOrangutan** [mathisdt/trackworktime#150]
- [ ] When 12 hour format is selected in system settings, time is shown in 24 hour format.
- [ ] Short date is fixed to American style format "mm/dd". It does not update to European format, like "dd.mm".
- [ ] Locale doesn't update immediately, requires app restart.
I could be wrong, but I think this should be reflected immediately. Maybe locale is cached somewhere.

![image](https://user-images.githubusercontent.com/5972966/112680775-12546680-8e6e-11eb-80c5-7a9f6f8e3de1.png)

**Comment by mathisdt** [comment]
You're right, the date and time presentation is not respecting the system settings. But as the app is only available in English, I as a user would not expect it to show any other date or time format than US (or GB). Yet, even if I use that as assumption, it's still not completely consistent: the time is shown in 24-hour format which a person from the USA probably wouldn't expect.

**Comment by hungerburg** [comment]
The principle of least surprise, if I understand correctly, suggests to display dates in the format of the locale chosen by the user in device settings. Especially so with dates, as not depending on language alone, e.g. "06/04" is not the same in US (4.6.) and GB (6.4.). No opinion here on how to cater for US time display.

**Comment by WildOrangutan** [comment]
Working on it 🚀

**Comment by trev0r-STA** [comment]
Hello
Beautiful app, Thank you for sharing it, but I've a little problem, all the hours are stored in the 24h format, but display in 12h format ?
What's wrong ?
Timezone set to Europe/Paris, if I open the backup :

type;time;task;comment;taskId;name;active;ordering;isdefault
in;2022-02-14T07:55:13+01:00;1;;1;Atelier;1;0;1;
out;2022-02-14T12:00:00+01:00;0;;;;;;;
in;2022-02-14T12:30:00+01:00;1;;1;Atelier;1;0;1;
out;2022-02-14T17:00:13+01:00;0;;;;;;;
in;2022-02-15T07:58:10+01:00;1;;1;Atelier;1;0;1;
out;2022-02-15T12:23:11+01:00;0;;;;;;;
in;2022-02-15T12:54:06+01:00;1;;1;Atelier;1;0;1;
out;2022-02-15T16:45:18+01:00;0;;;;;;;

And this is what I get:
![Screenshot_2022-02-15_20-48-01](https://user-images.githubusercontent.com/17881601/154139345-33379343-3d1a-4b61-b5cd-5fca5dcafd39.png)

If you need more information, ask me.
Thanks for your help

**Comment by mathisdt** [comment]
Hello @trev0r-STA,
it seems that your Android device is set to display the 12-hour format - at least on my test devices, I can't reproduce this without explicitly setting the time to 12-hour format. For details see below.

Here's what I see with my default "Germany/Berlin" locale:
![1_24_hour_format_in_TWT](https://user-images.githubusercontent.com/1932462/154149627-e42ba6ac-04d9-4d2d-a578-35d70fd716ee.jpg)

These are my time/date settings (now I switched my device to french so you may find the settings easier on your device):
![2_setting_default_is_24_hour](https://user-images.githubusercontent.com/1932462/154149794-bde12dda-92bb-435d-a195-8ddd881c9362.jpg)

Now I switch off "use default time format" **and also** the "use 24 hour format" like this:
![3_setting_changed_to_12_hour](https://user-images.githubusercontent.com/1932462/154150002-596b4349-23bc-45e3-aae2-d10e4e6340bd.jpg)

Afterwards I also see 12-hour formatted times in TWT (after I completely stopped and restarted the app):
![4_12_hour_format_in_TWT](https://user-images.githubusercontent.com/1932462/154150110-1135c5fd-87e1-43fd-921f-1257f0c7014d.jpg)

**Comment by trev0r-STA** [comment]
I don't have exactly the same parameters, but if I set Automatic time zone to OFF, and select the right time zone, that's OK, hours display correctly
![Screenshot_2022-02-15_22-17-56](https://user-images.githubusercontent.com/17881601/154151562-6c3c855a-6167-495f-800c-6181f3edc9a8.png)
 That's a little bit strange, no ?
Whatever, it works.
Thanks for your help and quick reply.
No translation in a near future ? If you need a french translator, I will be happy to help.
Have a nice end of day
